02_bubble/nicholascross/maint_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote(`[1 2 3 5]
+[1 2 3 5]
+`)
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main() %v", actual)
+	}
+}
+
+var testCases = map[string]struct {
+	input []int
+	want  []int
+}{
+	"sample1": {input: []int{1, 2, 3, 4, 5}, want: []int{1, 2, 3, 4, 5}},
+	"sample2": {input: []int{5, 4, 3, 2, 1}, want: []int{1, 2, 3, 4, 5}},
+	"sample3": {input: []int{1, 2, 4, 3, 5}, want: []int{1, 2, 3, 4, 5}},
+	"sample4": {input: []int{4, 3, 1, 5, 2}, want: []int{1, 2, 3, 4, 5}},
+	"sample5": {input: []int{5, 1, 2, 4, 3}, want: []int{1, 2, 3, 4, 5}},
```

Comment:
This test case does not seem to add anything that the previous cases don't already test.

Rather than naming your test cases sample1..6, if you gave them meaningful names of what is being tested, then you may find there is no name to give to this test as you've already done it. Or, perhaps I don't see what is special about this test and it is a useful test - if it had a useful name, I would see that.

So:
"sample1" -> "already sorted"
"sample2" -> "reverse order"
"sample3" -> "retain ends"
"sample4" -> "reorder all"
"sample5" -> ???
"sample6" -> "duplicate entries"

This is an informational-only comment and won't block approval.

Reply:
I separated sample6 into two cases: duplicates and negative numbers.

I also updated the test case names to be more meaningful.